generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/extensions/annotationExtensions.kt

```diff
@@ -32,7 +33,9 @@ internal fun KAnnotatedElement.getDeprecationReason(): String? = this.findAnnota
 
 internal fun KAnnotatedElement.isGraphQLIgnored(): Boolean = this.findAnnotation<GraphQLIgnore>() != null
 
-internal fun List<Annotation>.getUnionAnnotation(): GraphQLUnion? = this.filterIsInstance(GraphQLUnion::class.java).firstOrNull()
+internal fun List<Annotation>.getUnionAnnotation(): GraphQLUnion? = this.filterIsInstance(GraphQLUnion::class.java).firstOrNull() ?: this.map{ it.annotationClass.annotations.filterIsInstance(GraphQLUnion::class.java).firstOrNull() }.firstOrNull()
```

Comment:
I purposely didn't make this recursive as to only allow for `@GraphQLUnion` to be a single "layer" deep vs something like:
```kotlin
@GraphQLUnion(...)
annotation class Foo

@Foo
annotation class Bar

class Query {

@Bar
fun union: Any 
```